### PR TITLE
Feat/pow update parents

### PR DESCRIPTION
**Pull request by piotrm50**
# Description of change

PoW in IssuePayload updates timestamp and parents while looking for nonce.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Manual testing by running spammer and faucet requests.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by capossele — APPROVED** [review]

**Review by capossele — APPROVED** [review]

File: config.default.json [hunk 1]
```diff
@@ -111,6 +111,7 @@
     "difficulty": 22,
     "numThreads": 1,
     "timeout": "1m"
+
```
Comment (jonastheis) [inline config.default.json:114]: Should this be added here? `"parentsRefreshInterval": "300ms"`
Or do we just rely on the default value?

**Review by jonastheis — APPROVED** [review]

File: tools/docker-network/config.docker.json [hunk 2]
```diff
@@ -70,7 +70,8 @@
   "pow": {
     "difficulty": 2,
     "numThreads": 1,
-    "timeout": "10s"
+    "timeout": "10s",
```
Comment (apenzk) [inline tools/docker-network/config.docker.json:73]: lets use 60s as default

File: packages/tangle/messagefactory_test.go [hunk 3]
```diff
@@ -22,6 +23,7 @@ import (
 
 const (
 	targetPOW     = 10
+	powTimeout    = 10 * time.Second
```
Comment (apenzk) [inline packages/tangle/messagefactory_test.go:26]: lets use 60s here. with this new change we dont have problems with timestamps getting to old anymore

File: tools/docker-network/config.docker.json (hunk 2 above)
Comment (capossele) [reply tools/docker-network/config.docker.json:73]: here 10s is fine as it is for the docker-network where we set the difficulty to be only 2

File: packages/tangle/messagefactory_test.go (hunk 3 above)
Comment (capossele) [reply packages/tangle/messagefactory_test.go:26]: here 10s is fine as it is for the test where we set the difficulty to be only 2

File: config.default.json (hunk 1 above)
Comment (capossele) [reply config.default.json:114]: Yeah, we thought that by specifying that in the default value was easier so we don't care to set that here

**Pull request merged**